### DISCUSSION TITLE
Provide simple way to assign a name to test cases defined by InlineData attributes

### DISCUSSION THREAD

**Discussion by bhehe** [xunit/xunit#2532] [Ideas]
Coming from an NUnit background and learning the ins & outs of XUnit, I was looking into how I could assign names to each test case represented with an **[InlineData]** attribute and could not find a simple way to do so.  

I did see a topic relating to how things are serialized and ToString was used to get a name but I was looking for something that is simple and doesn't require crafting a 'test case wrapper' just to assign a name. 

i.e.

[InlineData(1, "SomeValue", "abc1234", **TestName =** "Primary Test Case")
[InlineData(2, "AlternateValue", "123xyz", **TestName =** "Some Other Test Case")

Is there already a way to do this and if not, is there anything on the roadmap to support it? Seems like a foundational thing to have and was surprised to find a lack of support for it.

Thanks!

**Comment by bradwilson** [comment]
This will be available in v3 for all data attributes (not just InlineData): https://github.com/xunit/xunit/blob/a9a5206405584517a7bb76c23dc270cee74464dc/src/xunit.v3.core/Sdk/DataAttribute.cs#L71-L77

In addition, the new [`ITheoryDataRow`](https://github.com/xunit/xunit/blob/a9a5206405584517a7bb76c23dc270cee74464dc/src/xunit.v3.core/ITheoryDataRow.cs) allows you to return individual data rows (from sources like `[MemberData]`) that include test names for individual data rows. Rather than returning `IEnumerable<object[]>`, you can return `IEnumerable<ITheoryDataRow>` (and the helper implementation [`TheoryDataRow`](https://github.com/xunit/xunit/blob/a9a5206405584517a7bb76c23dc270cee74464dc/src/xunit.v3.core/Sdk/TheoryDataRow.cs) makes it easy to return data rows of pre-enumerated data, decorated with a test name).

The following test illustrates the various ways to name tests in v3: https://github.com/xunit/xunit/blob/a9a5206405584517a7bb76c23dc270cee74464dc/src/xunit.v3.core.tests/Acceptance/Xunit3TheoryAcceptanceTests.cs#L759-L806

**Comment by bhehe** [comment]
That's great news! Thanks.

One thing I did note when looking at the example is the 'TestDisplayName' - was there anything that would preclude using the short-form of just 'TestName' for the sake of brevity?  No big deal but that was something that came to mind when looking at the example.

**Reply by bradwilson** [reply]
Internally we consider "name" and "display name" to be separate concepts, so we wanted to be explicit here as well.
